### PR TITLE
Feature: Update fetching structure to include domain, application, mongo (infra) and rest

### PR DESCRIPTION
**Pull request by WLefever-Cegeka**
In this PR:
* split `ldes-server-port-fetch` into `ldes-server-port-fetch` (domain module) and `ldes-server-port-fetch-application-layer` (application layer)
* moved all business logic from `ldes-server-port-fetch-rest` and `ldes-server-port-fetch-application-layer` into `ldes-server-port-fetch` (domain module)
* slimmed down `ldes-server-port-fetch-application-layer` and `ldes-server-port-fetch-rest`
* added exclusion on the use of  `ldes-server-port-fetch` (so that it's entities are safeguarded)
* added spring modulith to generate the module documentation (we cannot enforce modulith yet, since the other modules share entities)

Updates are all on `TreeNode`, left `EventStream` as it was, since there's still shared code with admin module.

Still to do:
* IT
* Mongock

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -29,6 +29,8 @@ COPY --from=app-stage ldes-server-port-ingest-rest/target/ldes-server-port-inges
 COPY --from=app-stage ldes-server-port-ingest/target/ldes-server-port-ingest-jar-with-dependencies.jar ./lib/
 COPY --from=app-stage ldes-server-port-fetch/target/ldes-server-port-fetch-jar-with-dependencies.jar ./lib/
 COPY --from=app-stage ldes-server-port-fetch-rest/target/ldes-server-port-fetch-rest-jar-with-dependencies.jar ./lib/
+COPY --from=app-stage ldes-server-port-fetch-application-layer/target/ldes-server-port-fetch-application-layer-jar-with-dependencies.jar ./lib/
```
Comment (Tomvbe) [inline Dockerfile:32]: I don't really understand why you added this new module.
Can't you use spring modulith and only activate the fetch profile and make sure that works? Then you don't have to update the entire server.

File: ldes-server-port-fetch-application-layer/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchapplication/mapper/MemberAllocatedMapper.java [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+package be.vlaanderen.informatievlaanderen.ldes.server.fetchapplication.mapper;
 
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.events.fragmentation.MemberAllocatedEvent;
-import be.vlaanderen.informatievlaanderen.ldes.server.fetching.entities.MemberAllocation;
+import be.vlaanderen.informatievlaanderen.ldes.server.fetchdomain.entities.MemberAllocation;
 import org.springframework.stereotype.Component;
 
 @Component
 public class MemberAllocatedMapper {
+	// TODO use MapStruct
```
Comment (Tomvbe) [inline ldes-server-port-fetch-application-layer/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchapplication/mapper/MemberAllocatedMapper.java:9]: Can you remove these todo's and replace them by a github issue?

File: ldes-server-port-fetch-application-layer/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchapplication/eventhandler/EventStreamCreatedHandlerFetch.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		this.eventStreamRepository = eventStreamRepository;
+		this.hostname = hostName;
+	}
+
+	@EventListener
+	public void handleEventStreamCreatedEvent(EventStreamCreatedEvent event) {
+		eventStreamRepository.saveEventStream(getEventStream(event));
+	}
+
+	// TODO use MapStruct
```
Comment (Yalz) [inline ldes-server-port-fetch-application-layer/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchapplication/eventhandler/EventStreamCreatedHandlerFetch.java:30]: remove todo or make issue 😄

File: ldes-server-port-fetch-application-layer/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchapplication/services/TreeNodeFactoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+			EventStreamRepository eventStreamRepository, DcatViewService dcatViewService) {
+		this.fragmentRepository = fragmentRepository;
+		this.allocationRepository = allocationRepository;
+		this.memberRepository = memberRepository;
+		this.shaclRepository = shaclRepository;
+		this.eventStreamRepository = eventStreamRepository;
+		this.dcatViewService = dcatViewService;
+	}
+
+	// TODO use MapStruct
```
Comment (Yalz) [inline ldes-server-port-fetch-application-layer/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchapplication/services/TreeNodeFactoryImpl.java:43]: remove or make issue

File: ldes-server-port-fetch/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchdomain/valueobjects/EventStreamInfo.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+			List<Statement> dcatStatements, EventStreamProperties eventStreamProperties) {
+		this.treeNodeIdentifier = treeNodeIdentifier;
+		this.eventStreamIdentifier = eventStreamIdentifier;
+		this.shacl = shacl;
+		this.isView = isView;
+		this.dcatStatements = dcatStatements;
+		this.eventStreamProperties = eventStreamProperties;
+	}
+
+	// TODO use polymorphism
```
Comment (Yalz) [inline ldes-server-port-fetch/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchdomain/valueobjects/EventStreamInfo.java:31]: remove or create issue

File: ldes-server-port-fetch/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchdomain/constants/RdfConstants.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.fetchdomain.constants;
+
+import org.apache.jena.rdf.model.Property;
+
+import static org.apache.jena.rdf.model.ResourceFactory.createProperty;
+
+public class RdfConstants {
```
Comment (Tomvbe) [inline ldes-server-port-fetch/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchdomain/constants/RdfConstants.java:7]: I really don't think we should be duplicating these constants. Sooner or later they will get out of sync and will result in bugs.
Do we really want to define some of these constants such as TREE in every module? That means that other modules can give it another String, which is something you would not expect when reading the code and can result in false assertions and again result in bugs.

File: ldes-server-port-fetch/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchdomain/services/PrefixAdderImpl.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.jena.shared.PrefixMapping;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+//TODO extract to util
+public class PrefixAdderImpl implements PrefixAdder {
```
Comment (Tomvbe) [inline ldes-server-port-fetch/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetchdomain/services/PrefixAdderImpl.java:14]: I would not duplicate this class either.. This is full of complex logic and will definitely result in a bug sooner than later.

**Pull request closed without merging**